plugins/ingest-attachment/src/main/java/org/elasticsearch/ingest/attachment/AttachmentProcessor.java

```diff
@@ -161,59 +191,99 @@ String getTargetField() {
             int indexedChars = readIntProperty(TYPE, processorTag, config, "indexed_chars", NUMBER_OF_CHARS_INDEXED);
             boolean ignoreMissing = readBooleanProperty(TYPE, processorTag, config, "ignore_missing", false);
 
-            final Set<Property> properties;
-            if (properyNames != null) {
-                properties = EnumSet.noneOf(Property.class);
-                for (String fieldName : properyNames) {
-                    try {
-                        properties.add(Property.parse(fieldName));
-                    } catch (Exception e) {
-                        throw newConfigurationException(TYPE, processorTag, "properties", "illegal field option [" +
-                            fieldName + "]. valid values are " + Arrays.toString(Property.values()));
+            final Set<ReservedProperty> reservedProperties;
+            final Set<String> properties = new HashSet<>();
+            if (propertyNames != null) {
+                reservedProperties = EnumSet.noneOf(ReservedProperty.class);
+                for (String fieldName : propertyNames) {
+                    // We build the regex automaton we will use to extract raw metadata
+                    CharacterRunAutomaton automaton = new CharacterRunAutomaton(simpleMatchToAutomaton(fieldName));
+
+                    // Let's see if available reserved properties match what the user asked for
+                    EnumSet<ReservedProperty> allReservedProperties = EnumSet.allOf(ReservedProperty.class);
+                    for (ReservedProperty reservedProperty : allReservedProperties) {
+                        if (automaton.run(reservedProperty.key)) {
+                            logger.trace("found a reserved property: [{}]", reservedProperty.key);
+                            reservedProperties.add(reservedProperty);
+                        } else if (fieldName.equals(reservedProperty.name().toLowerCase(Locale.ROOT))) {
+                            deprecationLogger.deprecated("[{}] should be replaced with [{}]", reservedProperty.toLowerCase(),
+                                reservedProperty.key);
+                            reservedProperties.add(reservedProperty);
+                        }
                     }
+
+                    // And add the property
+                    properties.add(fieldName);
                 }
             } else {
-                properties = DEFAULT_PROPERTIES;
+                reservedProperties = DEFAULT_PROPERTIES;
+                logger.trace("no properties provided, falling back to default reserved properties: [{}]", reservedProperties);
             }
 
-            return new AttachmentProcessor(processorTag, field, targetField, properties, indexedChars, ignoreMissing);
+            return new AttachmentProcessor(processorTag, field, targetField, reservedProperties, properties, indexedChars, ignoreMissing);
         }
     }
 
-    enum Property {
+    enum ReservedProperty {
```

Comment:
do you really need this enum? Isnt it just a list of strings, which you can also check with an underscore at the beginning at the end?